### DISCUSSION TITLE
Ability to set an end time?

### DISCUSSION THREAD

**Discussion by chrisgurney** [SelfControlApp/selfcontrol#686] [Ideas]
_This app has saved on many projects I've been working on. THANK YOU and the team for your contributions, keeping this free, and for keeping me focussed!_

The only thing that I'd find real handy is a means of setting an end time, like the end of my work day. At the moment I have to calculate when that is and choose the hours to block; it would be far easier if I could just tell the block to end at 11pm, for example. Any chance that's going to be in 4.0?

Thanks again!

**Comment by slambert** [comment]
@andreasgrill created something that may help: https://github.com/andreasgrill/auto-selfcontrol

**Comment by cstigler** [comment]
@chrisgurney I like this idea :) I don't think it really needs to be a new input field, I imagine we could just change the label to say "3 hours 30 minutes (ends at 4:11pm)" - was that what you were thinking?

(might make it kinda long/clunky, but perhaps there's a better way to display the end time in the UI?)

**Reply by chrisgurney** [reply]
@cstigler I'd be happy with that, for sure!

If you're looking for inspiration, take a look at this timer app; I like the simplicity of the UI: https://github.com/michaelvillar/timer-app

**Reply by karbassi** [reply]
> @cstigler I'd be happy with that, for sure!
> 
> If you're looking for inspiration, take a look at this timer app; I like the simplicity of the UI: https://github.com/michaelvillar/timer-app

Speaking of timer-app, we're dealing with a [bug in Big Sur](https://github.com/michaelvillar/timer-app/issues/95) that we could use an MacOS dev. ;)
